examples/image-blur/main.zig

```diff
@@ -0,0 +1,290 @@
+
+    const fullscreen_quad_vs_module = core.device.createShaderModule(&.{
+        .label = "my fullscreen quad vertex shader",
+        .code = .{ .wgsl = @embedFile("fullscreen_textured_quad.wgsl") },
+    });
+
+    const fullscreen_quad_fs_module = core.device.createShaderModule(&.{
+        .label = "my fullscreen quad fragment shader",
+        .code = .{ .wgsl = @embedFile("fullscreen_textured_quad.wgsl") },
+    });
+
+    const blend = gpu.BlendState{
+        .color = .{
+            .operation = .add,
+            .src_factor = .one,
+            .dst_factor = .zero,
+        },
+        .alpha = .{
+            .operation = .add,
+            .src_factor = .one,
+            .dst_factor = .zero,
+        },
+    };
+
+    const color_target = gpu.ColorTargetState{
+        .format = core.swap_chain_format,
+        .blend = &blend,
+        .write_mask = gpu.ColorWriteMask.all,
+    };
+
+    const fragment_state = gpu.FragmentState{
+        .module = fullscreen_quad_fs_module,
+        .entry_point = "frag_main",
+        .targets = &.{color_target},
+        .constants = null,
+    };
+
+    const fullscreen_quad_pipeline_descriptor = gpu.RenderPipeline.Descriptor{
+        .layout = null,
+        .fragment = &fragment_state,
+        .vertex = .{
+            .module = fullscreen_quad_vs_module,
+            .entry_point = "vert_main",
+            .buffers = null,
+        },
+    };
+
+    const fullscreen_quad_pipeline = core.device.createRenderPipeline(&fullscreen_quad_pipeline_descriptor);
+
+    const sampler = core.device.createSampler(&.{
+        .mag_filter = .linear,
+        .min_filter = .linear,
+    });
+
+    const img = try zigimg.Image.fromMemory(core.allocator, @embedFile("../assets/gotta-go-fast.png"));
+    defer img.deinit();
+
+    const img_size = gpu.Extent3D{ .width = @intCast(u32, img.width), .height = @intCast(u32, img.height) };
+
+    const cube_texture = core.device.createTexture(&.{
+        .size = img_size,
+        .format = .rgba8_unorm,
+        .usage = .{
+            .texture_binding = true,
+            .copy_dst = true,
+            .render_attachment = true,
+        },
+    });
+
+    const data_layout = gpu.Texture.DataLayout{
+        .bytes_per_row = @intCast(u32, img.width * 4),
+        .rows_per_image = @intCast(u32, img.height),
+    };
+
+    switch (img.pixels.?) {
+        .Rgba32 => |pixels| queue.writeTexture(&.{ .texture = cube_texture }, &data_layout, &img_size, zigimg.color.Rgba32, pixels),
+        .Rgb24 => |pixels| {
+            const data = try rgb24ToRgba32(core.allocator, pixels);
+            defer data.deinit(core.allocator);
+            queue.writeTexture(&.{ .texture = cube_texture }, &data_layout, &img_size, zigimg.color.Rgba32, data.Rgba32);
+        },
+        else => @panic("unsupported image color format"),
+    }
+
+    var textures: [2]gpu.Texture = undefined;
+    for ([_]i32{ 0, 1 }) |_, i| {
```

Comment:
Change this to `for (textures) |_, i| {`, then it's not possible to work on the wrong amount of items.